src/strategies/saddle-finance/index.ts

```diff
@@ -0,0 +1,148 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+import fetch from 'cross-fetch';
+import { vestingContractAddrs } from './vestingContractAddrs';
+
+export const author = 'saddle-finance';
+export const version = '0.1.0';
+
+const SDLTokenAddress = '0xf1Dc500FdE233A4055e25e5BbF516372BC4F6871';
+const RetroRewardsContract = '0x5DCA270671935cf3dF78bd8373C22BE250198a03';
+
+const abi = [
+  'function balanceOf(address) external view returns (uint256)',
+  'function beneficiary() external view returns (address)',
+  'function vestings(address) external view returns (bool isVerified, uint120 totalAmount, uint120 released)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const remappedMerkleDataRes = await fetch(
+    'https://gateway.pinata.cloud/ipfs/QmV73GEaijyiBFHu1vRdZBFffoCHaXYWG5SpurbEgr4VK6',
+    {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      }
+    }
+  );
+  const remappedMerkleData = await remappedMerkleDataRes.json();
+
+  const userWalletBalanceResponse = await multicall(
+    network,
+    provider,
+    abi,
+    addresses.map((address: any) => [
+      SDLTokenAddress,
+      'balanceOf',
+      [address.toLowerCase()]
+    ]),
+    { blockTag }
+  );
```

Comment:
Can you try moving all 4 multicall promises into one `promise.all` so all multicalls will happen at the same time and could save us some time  ( as they are not dependent on each other)

make sure to run this in your local
```bash
npm test --strategy=saddle-finance
```

Reply:
Thanks, seems to be consistently running @ under 10s locally now, hopefully, the same holds on CI.